sdks/python/apache_beam/io/filebasedsource_test.py

```diff
@@ -41,6 +41,14 @@
 class LineSource(FileBasedSource):
 
   def read_records(self, file_name, range_tracker):
+    current = -1
+
+    def _callback(stop_position):
+      return FileBasedSource.remaining_split_points_helper(
+          stop_position, current, range_tracker.done())
```

Comment:
The range tracker lock is a different lock than the source lock. You need them both.